openshift-201/rh201-pod-auto-scale.md

```diff
@@ -0,0 +1,215 @@
+# Autoscaling Pods
+## Objectives
+
+After completing this section you should have a better understanding of Horizontal Pod Autoscalers (HPA), Vertical Pod Autoscalers (VPA), and Pod Disruption Budgets (PDB). And go through the steps to configure and observe these resources.
+
+## Horizontal Pod Autoscalers
+
+OpenShift can autoscale a deployment based on current load on the application pods, by means of a HorizontalPodAutoscaler resource type.
+
+A horizontal pod autoscaler (HPA) resource uses performance metrics collected by the OpenShift Metrics subsystem. The Metrics subsystem comes pre-installed in OpenShift 4.
+
+You can also create an HPA for DeploymentConfig, ReplicaSet, ReplicationController, or StatefulSet objects.
+
+This command will create a horizontal pod autoscaler resource that changes the number of replicas on the hello-world-nginx deployment to keep its pods under 80% of their total requested CPU usage.
+
+```
+oc autoscale deployment/hello-world-nginx --min 1 --max 5 --cpu-percent 80
+```
+
+The maximum and minimum values for the horizontal pod autoscaler resource serve to accommodate bursts of load and avoid overloading the OpenShift cluster. If the load on the application changes too quickly, then it might be advisable to keep a number of spare pods to cope with sudden bursts of user requests. Conversely, too many pods can use up all cluster capacity and impact other applications sharing the same OpenShift cluster.
```

Comment:
maybe also mention how to find a good balance on target utilization percentage? For example, if set to 80%, that means the rest 20% of resource is sufficient to handle the load increase before new pod gets ready.

Reply:
Good call! Updated!